### PR TITLE
fix(svm): root bundles over cctp

### PR DESCRIPTION
**Pull request by Reinis-FRP**
We require ability to relay and emergency delete root bundles over CCTP.

Fixes: https://linear.app/uma/issue/ACX-3029/instructionsadminrs-remote-relay-emergency-delete-root-bundle

### CODE REVIEW

File: programs/svm-spoke/src/instructions/admin.rs [hunk 1]
```diff
@@ -257,17 +259,21 @@ pub fn relay_root_bundle(
 #[instruction(root_bundle_id: u32)]
 pub struct EmergencyDeleteRootBundle<'info> {
     #[account(
-        mut,
         constraint = is_local_or_remote_owner(&signer, &state) @ CustomError::NotOwner
     )]
     pub signer: Signer<'info>,
 
+    #[account(mut)]
+    // We do not restrict who can receive lamports from closing root_bundle account as that would require storing the
```
Comment (chrismaree) [inline programs/svm-spoke/src/instructions/admin.rs:267]: makes sense, as long as it's authenticated with `is_local_or_remote_owner` we don't really mind.

File: programs/svm-spoke/src/instructions/handle_receive_message.rs [hunk 2]
```diff
@@ -82,6 +82,18 @@ fn translate_message(data: &Vec<u8>) -> Result<Vec<u8>> {
             (origin_token, destination_chain_id, enabled)
                 .encode_instruction_data("global:set_enable_route")
         }
+        s if s == utils::encode_solidity_selector("relayRootBundle(bytes32,bytes32)") => {
```
Comment (chrismaree) [inline programs/svm-spoke/src/instructions/handle_receive_message.rs:85]: great! we can now test this on testnet as well over cctp

File: programs/svm-spoke/src/instructions/admin.rs (hunk 1 above)
Comment (Reinis-FRP) [reply programs/svm-spoke/src/instructions/admin.rs:267]: In remote authentication this checks only that this is coming from HubPool via CCTP, but we do not use permissioned caller to receive the message on Solana side (though CCTP has that option). So there might be a case that someone frontruns receiving message on Solana and gets all lamports from the closing of root bundle account. But I think this is rare enough when we would use this emergency method and it should be no concern that someone else could gain some extra lamports.

Comment (md0x) [reply programs/svm-spoke/src/instructions/admin.rs:267]: Makes sense. Anyone can "finalize" this messages

File: test/svm/SvmSpoke.HandleReceiveMessage.ts [hunk 3]
```diff
@@ -406,4 +409,168 @@ describe("svm_spoke.handle_receive_message", () => {
...
+    // Remaining accounts specific to RelayRootBundle.
+    const rootBundleId = (await program.account.state.fetch(state)).rootBundleId;
+    const rootBundleIdBuffer = Buffer.alloc(4);
+    rootBundleIdBuffer.writeUInt32LE(rootBundleId);
+    const seeds = [Buffer.from("root_bundle"), state.toBuffer(), rootBundleIdBuffer];
+    const [rootBundle] = web3.PublicKey.findProgramAddressSync(seeds, program.programId);
+    // Same 3 remaining accounts passed for HandleReceiveMessage context.
+    const relayRootBundleRemainingAccounts = remainingAccounts.slice(0, 3);
+    // payer in self-invoked SetEnableRoute.
+    relayRootBundleRemainingAccounts.push({
```
Comment (md0x) [inline test/svm/SvmSpoke.HandleReceiveMessage.ts:439]: Does order matter here?

**Review by md0x — APPROVED** [review]
Looks good!

File: test/svm/SvmSpoke.HandleReceiveMessage.ts (hunk 3 above)
Comment (Reinis-FRP) [reply test/svm/SvmSpoke.HandleReceiveMessage.ts:439]: Regular accounts have named props, so the client can reorder them as needed, but remaining accounts is plain array, so I believe the ordering here matters.

**Review by chrismaree — APPROVED** [review]

**Pull request merged**